lib/smooth/application/io/spi/BME280SPI.cpp

```diff
@@ -0,0 +1,256 @@
+                                             spi_interface);
+
+        // for spi write the bit 7 of register address must be zero
+        datagram.at(0) = datagram.at(0) & 0x7F;  // modify 0xF2 to 0x72
+        datagram.at(2) = datagram.at(2) & 0x7F;  // modify 0xF4 to 0x74
+        datagram.at(4) = datagram.at(4) & 0x7F;  // modify 0xF5 to 0x75
+
+        return write(datagram);
+    }
+
+    bool BME280SPI::read_configuration(BME280Core::SensorMode& mode,
+                                       BME280Core::OverSampling& humidity,
+                                       BME280Core::OverSampling& pressure,
+                                       BME280Core::OverSampling& temperature,
+                                       BME280Core::StandbyTimeMS& standby,
+                                       BME280Core::FilterCoeff& filter,
+                                       BME280Core::SpiInterface& spi_interface)
+    {
+        // The spi rx buffer need to be length of multiples of 32 bits to avoid heap corruption.c
+        alignas(4) core::util::FixedBuffer<uint8_t, 4> ctrl_hum_data;
+        alignas(4) core::util::FixedBuffer<uint8_t, 4> ctrl_meas_data;
+        alignas(4) core::util::FixedBuffer<uint8_t, 4> config_data;
+
+        auto res = read(BME280Core::CTRL_HUM_REG, ctrl_hum_data.data(), 2) &&
+                   read(BME280Core::CTRL_MEAS_REG, ctrl_meas_data.data(), 2) &&
+                   read(BME280Core::CONFIG_REG, config_data.data(), 2);
+
+        if (res)
+        {
+            bme280_core.get_configuration_settings(ctrl_hum_data[1],
+                                                   ctrl_meas_data[1],
+                                                   config_data[1],
+                                                   mode,
+                                                   humidity,
+                                                   pressure,
+                                                   temperature,
+                                                   standby,
+                                                   filter,
+                                                   spi_interface);
+        }
+
+        return res;
+    }
+
+    bool BME280SPI::read_trimming_parameters()
+    {
+        if (!trimming_read)
+        {
+            // The spi rx buffer need to be length of multiples of 32 bits to avoid heap corruption.
+            alignas(4) core::util::FixedBuffer<uint8_t, 32> calib00_calib25_data;    // 0x88-0xA1
+            alignas(4) core::util::FixedBuffer<uint8_t, 8> calib26_calib32_data;     // 0xE1-0xE7
+            core::util::FixedBuffer<uint8_t, 32> calibration_data;
+
+            trimming_read = read(BME280Core::CALIB00_REG, calib00_calib25_data.data(), 27) &&
+                            read(BME280Core::CALIB26_REG, calib26_calib32_data.data(), 8);
+
+            if (trimming_read)
+            {
+                // don't copy dummy byte but copy 0x88-0x9F into calibration_data
```

Comment:
I admit I haven't read the SPI spec for BME280 - what's the dummy byte about? Can you add a short explanation about it for future reference?